### PR TITLE
feat: CRUD endpoints for alarm feature

### PR DESCRIPTION
**Pull request by thinhle19**
Implementation of endpoints:

POST /alarm-conditions
GET /alarm-conditions
GET /alarm-conditions/{alarmConditionId}
GET /sensor-configurations?blueprintType=string&blueprintName=string
Some major changes

Added Column Type to Blueprint Entity
Deleted Column Name of Alarm Condition Entity

### CODE REVIEW

File: core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/AlarmConditionController.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController()
```
Comment (oHTGo) [inline core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/AlarmConditionController.java:19]:
```suggestion
@RestController
```

File: core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/AlarmConditionController.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  private final AlarmConditionService alarmConditionService;
+  private final AlarmConditionPersistenceService alarmConditionPersistenceService;
+
+  @PostMapping()
+  public ApiResponse<AlarmCondition> create(@Valid @RequestBody CreateAlarmConditionCommand body) {
+    AlarmCondition result = alarmConditionService.createAlarmCondition(body);
+    return ApiResponse.success(result);
+  }
+
+  @GetMapping()
```
Comment (oHTGo) [inline core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/AlarmConditionController.java:33]:
```suggestion
  @GetMapping
```

File: core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/AlarmConditionController.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+@RestController()
+@RequestMapping("alarm-conditions")
+@RequiredArgsConstructor
+public class AlarmConditionController {
+  private final ModelMapper mapper;
+  private final AlarmConditionService alarmConditionService;
+  private final AlarmConditionPersistenceService alarmConditionPersistenceService;
+
+  @PostMapping()
```
Comment (oHTGo) [inline core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/AlarmConditionController.java:27]:
```suggestion
  @PostMapping
```

File: core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/SensorConfigurationController.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("sensor-configurations")
+@RequiredArgsConstructor
+public class SensorConfigurationController {
+  private final SensorConfigurationPersistenceService sensorConfigurationPersistenceService;
+
+  @GetMapping()
```
Comment (oHTGo) [inline core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/SensorConfigurationController.java:18]:
```suggestion
  @GetMapping
```

File: core/core-api/src/main/java/com/hbc/pms/core/api/config/plc/PlcDataSource.java [hunk 5]
```diff
@@ -16,7 +16,7 @@ public class PlcDataSource implements HbcScrapeJobDataSource {
 
   @Override
   public Map<String, String> getTags() {
-    return blueprintService.getAllAddresses().stream()
+    return blueprintService.getAllMonitoringAddresses().stream()
```
Comment (oHTGo) [inline core/core-api/src/main/java/com/hbc/pms/core/api/config/plc/PlcDataSource.java:19]: Please revert it to fetch monitoring addresses & alarm addresses

File: local.properties [hunk 6]
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
```
Comment (oHTGo) [inline local.properties:1]: Please remove it

**Review by oHTGo — COMMENTED** [review]
Added some comments

File: core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/request/SearchBlueprintCommand.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+public class SearchBlueprintCommand {
+  private BlueprintType blueprintType;
+  private String blueprintName;
```
[the remark this answers is not in the document]
Comment (thinhle19) [reply core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/request/SearchBlueprintCommand.java:13]: The dto is used for sensor-config endpoint. So specifically stating out `blueprintType` and `blueprintName` might be better isn't it?
image

File: core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/request/CreateAlarmConditionCommand.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      return min < max;
+    }
+
+    return true;
+  }
+
+  @Data
+  public static class CreateAlarmActionCommand {
+    private AlarmActionType type;
+    private Set<String> recipientIds;
```
[the remark this answers is not in the document]
Comment (thinhle19) [reply core/core-api/src/main/java/com/hbc/pms/core/api/controller/v1/request/CreateAlarmConditionCommand.java:56]: I think using String will be better for extending later, để zậy nha

**Comment by huybui38** [comment]
merge di plaASE

**Pull request merged**